### DISCUSSION TITLE
How to upload a media with OAuth2 authentication?

### DISCUSSION THREAD

**Discussion by dev-bre** [plhery/node-twitter-api-v2#416] [General]
I am using OAuth2 authentication, and even though I see that the uploadMedia function is only available for V1, it should still be possible to use it. 

Unfortunately I  keep getting 403 Forbidden error when I call it.

any idea why? 

Please don't tell me it is only available with OAuth1.1 auth... please!

**Comment by solojungle** [comment]
Having the same issue here. Wondering if anyone's solved this?

**Reply by solojungle** [reply]
Solution for me:

* User signs up for my app, so I used the v2 client to tweet i.e:
```
export const client = new TwitterApi({
	clientId: env.TWITTER_CLIENT_ID,
	clientSecret: env.TWITTER_CLIENT_SECRET,
});
```

But, uploadMedia does not work with v2, you have to use v1 to upload.
```
export const v1client = new TwitterApi({
	appKey: "",
	appSecret: "",
	accessToken: "",
	accessSecret: "",
});

```

Hope this helps anyone else out!

**Reply by cybertronprime** [reply]
So just  we upload the media using v1 and then use the media id in the v2 , but i want to understand  how do we do it on behalf of user
